cmd/web-handlers.go

```diff
@@ -74,13 +74,111 @@ type ServerInfoRep struct {
 	UIVersion       string `json:"uiVersion"`
 }
 
+// ToKeyValuer interface wraps ToKeyValue method that allows types to
+// marshal their values as a map of structure member names to their
+// values, as strings
+type ToKeyValuer interface {
+	ToKeyValue() map[string]string
+}
+
+// ToKeyValue implementation for WebGenericArgs
+func (args *WebGenericArgs) ToKeyValue() map[string]string {
+	return map[string]string{}
+}
+
+// ToKeyValue implementation for MakeBucketArgs
+func (args *MakeBucketArgs) ToKeyValue() map[string]string {
+	return map[string]string{
+		"BucketName": args.BucketName,
+	}
+}
+
+// ToKeyValue implementation for RemoveBucketArgs
+func (args *RemoveBucketArgs) ToKeyValue() map[string]string {
+	return map[string]string{
+		"BucketName": args.BucketName,
+	}
+}
+
+// ToKeyValue implementation for ListObjectsArgs
+func (args *ListObjectsArgs) ToKeyValue() map[string]string {
+	return map[string]string{
+		"BucketName": args.BucketName,
+		"Prefix":     args.Prefix,
+		"Marker":     args.Marker,
+	}
+}
+
+// ToKeyValue implementation for RemoveObjectArgs
+func (args *RemoveObjectArgs) ToKeyValue() map[string]string {
+	objects, err := json.Marshal(args.Objects)
+	if err != nil {
+		// NB this can only happen when we can't marshal a Go
+		// slice to its json representation.
+		objects = []byte("[]")
+	}
+	return map[string]string{
+		"BucketName": args.BucketName,
+		"Objects":    string(objects),
```

Comment:
can these constants be named as they are repeated quite a few times?

Reply:
@poornas, that's a good point. It made realise that I can add a type (alias) to the `map[string]string` which will have methods like `SetBucket`, `SetObject` and `SetPrefix` initially, abstracting the keys. Let me update this PR.

Reply:
@poornas done